Content/Items/Vitric/Weapons.RecursiveFocus.cs

```diff
@@ -0,0 +1,905 @@
+        }
+
+        public override void SetDefaults()
+        {
+            Item.CloneDefaults(ItemID.QueenSpiderStaff);
+            Item.sentry = true;
+            Item.damage = 14;
+            Item.DamageType = DamageClass.Summon;
+            Item.mana = 15;
+            Item.knockBack = 0f;
+
+            Item.rare = ItemRarityID.Orange;
+            Item.value = Item.sellPrice(gold : 2);
+
+            Item.UseSound = SoundID.Item25;
+            Item.useStyle = ItemUseStyleID.HoldUp;
+
+            Item.shoot = ModContent.ProjectileType<RecursiveFocusProjectile>();
+            Item.shootSpeed = 1f;
+
+            Item.useTime = Item.useAnimation = 35;
+        }
+
+        public override bool Shoot(Player player, EntitySource_ItemUse_WithAmmo source, Vector2 position, Vector2 velocity, int type, int damage, float knockback)
+        {
+            if (player.ownedProjectileCounts[Item.shoot] > 0)
+            {
+                for (int i = 0; i < Main.maxProjectiles; i++)
+                {
+                    Projectile proj = Main.projectile[i];
+                    if (proj.active && proj.type == Item.shoot && proj.owner == player.whoAmI)
+                    {
+                        proj.Kill();
+                    }
+                }
+            }
+            Projectile.NewProjectileDirect(source, player.Center, velocity, type, damage, knockback, player.whoAmI).originalDamage = Item.damage;
+            player.UpdateMaxTurrets();
+            return false;
+        }
+    }
+
+    public class RecursiveFocusProjectile : ModProjectile
+    {
+        public Player owner => Main.player[Projectile.owner];
+
+        public override string Texture => AssetDirectory.VitricItem + Name;
+
+        public int modeSwitchTransitionTimer;
+
+        public int modeSwitchCooldown;
+
+        public int flashTimer;
+
+        public int trailMult = 5;
+
+        public int trailMult2 = 3;
+
+        public int timeSpentOnTarget;
+
+        public int timeSpentOnMultiTargetOne;
+        public int timeSpentOnMultiTargetTwo;
+
+        public Vector2 targetCenter;
+
+        public Vector2 multiTargetCenterOne;
+        public Vector2 multiTargetCenterTwo;
+
+        public bool rightClickMode;
+
+        public bool foundTarget;
+
+        public bool foundMultiTargetOne;
+        public bool foundMultiTargetTwo;
+
+        public bool flashing;
+
+        public NPC oldTargetNPC;
+        public NPC targetNPC;
+
+        public NPC multiTargetNPCOne;
+        public NPC multiTargetNPCTwo;
+
+        private List<Vector2> cache;
+        private Trail trail;
+
+        private List<Vector2> cache2;
+        private Trail trail2;
+
+        private List<Vector2> multiCacheOne;
```

Comment:
This kind of pattern looks a bit nasty and hardcoded, would it possibly make more sense to have the lasers be their own individual projectiles?